server/sockets/list.js

```diff
@@ -348,6 +384,156 @@ const emitRemoveMemberOnLeaveCohort = socket =>
+                    .to(socketId)
+                    .emit(ListActionTypes.LEAVE_ON_TYPE_CHANGE_SUCCESS, {
+                      cohortId,
+                      isCohortMember,
+                      listId,
+                      type
+                    });
+                }
+              }
+
+              if (dashboardClients.has(userId)) {
+                const { socketId } = dashboardClients.get(userId);
+
+                socket.broadcast
+                  .to(socketId)
+                  .emit(ListActionTypes.DELETE_SUCCESS, listId);
+              }
+
+              if (cohortClients.has(userId)) {
+                const { viewId, socketId } = cohortClients.get(userId);
+
+                if (viewId === cohortId.toString()) {
+                  socket.broadcast
+                    .to(socketId)
+                    .emit(ListActionTypes.DELETE_SUCCESS, listId);
+                }
+              }
+            });
+          }
+
+          viewersIds.forEach(id => {
+            const userId = id.toString();
+
+            if (dashboardClients.has(userId)) {
+              const { socketId } = dashboardClients.get(userId);
+
+              socket.broadcast
+                .to(socketId)
+                .emit(ListActionTypes.FETCH_META_DATA_SUCCESS, {
+                  [list._id]: { ...responseWithList(list, userId) }
+                });
+            }
+
+            if (cohortClients.has(userId)) {
+              const { viewId, socketId } = cohortClients.get(userId);
+
+              if (viewId === cohortId.toString()) {
+                socket.broadcast
+                  .to(socketId)
+                  .emit(ListActionTypes.FETCH_META_DATA_SUCCESS, {
+                    [list._id]: { ...responseWithList(list, userId) }
+                  });
+              }
+            }
+          });
+
+          socket.broadcast
+            .to(`sack-${listId}`)
+            .emit(ListActionTypes.CHANGE_TYPE_SUCCESS, data);
+        }
+      });
+  });
+
+const removeListMember = (
+  socket,
+  dashboardClients,
+  listClients,
+  cohortClients
+) =>
+  socket.on(ListActionTypes.REMOVE_MEMBER_SUCCESS, data => {
+    const { listId, userId } = data;
+
+    // Broadcast to that removed user if he is at dashboard view
+    if (dashboardClients.has(userId)) {
+      const { socketId } = dashboardClients.get(userId);
+
+      socket.broadcast
+        .to(socketId)
+        .emit(ListActionTypes.DELETE_SUCCESS, listId);
+    }
+
+    if (listClients.has(userId)) {
+      List.findById(listId)
+        .lean()
+        .exec()
+        .then(doc => {
+          if (doc) {
+            const { socketId, viewId } = listClients.get(userId);
+            const { cohortId } = doc;
+            const data = { listId };
+
+            if (viewId === listId) {
+              if (cohortId) {
+                data.cohortId = cohortId;
+              }
+
+              // Broadcast to that removed user if he is on the list view
```

Comment:
remove comment you use listClients and compare viewId === listId so it's clear that user is on the list view